### PR TITLE
Add WHALE and Register migaloo-archway

### PR DESCRIPTION
**Pull request by calabiyau87**
This PR adds WHALE, Native token of the Migaloo blockchain, to multiple chain assetlists:

Terra2
Terra
Archway
Juno
Chihuahua
Injective

Additionally it also registers migaloo-archway connection in _IBC

### CODE REVIEW

File: _IBC/archway-migaloo.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "../ibc_data.schema.json",
+  "chain_1": {
+    "chain_name": "migaloo",
```
Comment (JeremyParish69) [inline _IBC/archway-migaloo.json:4]: chain_1 must be `archway` in this case, not `migaloo` (to follow alphabetical order)

Comment (calabiyau87) [reply _IBC/archway-migaloo.json]: @JeremyParish69 Corrected

**Comment by calabiyau87** [comment]
I did a test transfer from Terra (LUNC) to Migaloo:
https://finder.terra.money/classic/tx/B349394425A91FFABB17E4942F496A264B9616857B75DB4D4D4C6451D20F7E99

The channels don't match what is already registerd in _IBC/migaloo-terra.json

Do I need to register another IBC path or update the existing?

**Comment by JeremyParish69** [comment]
> I did a test transfer from Terra (LUNC) to Migaloo: https://finder.terra.money/classic/tx/B349394425A91FFABB17E4942F496A264B9616857B75DB4D4D4C6451D20F7E99
> 
> The channels don't match what is already registerd in _IBC/migaloo-terra.json
> 
> Do I need to register another IBC path or update the existing?

Only one transfer/transfer channel may be registered, but none exists currently, so this transfer/transfer channel may be added to the existing file (above the current channel, because that's has a wasm channel). It would be required to registered WHALE on Terra Classic.

File: terra/assetlist.json [hunk 2]
```diff
@@ -6359,6 +6359,58 @@
...
+      "name": "Migaloo",
+      "display": "whale",
+      "symbol": "WHALE",
+      "traces": [
+        {
+          "type": "ibc",
+          "counterparty": {
+            "chain_name": "migaloo",
+            "base_denom": "uwhale",
+            "channel_id": "channel-114"
```
Comment (JeremyParish69) [inline terra/assetlist.json:6389]: This looks to be incorrect

**Comment by calabiyau87** [comment]
So what is the schema to add an additional channel because there is a new client id, channel, and connection id for the transfer/transfer between migaloo<>terra
Screenshot 2024-12-05 115234
Screenshot 2024-12-05 115309
Screenshot 2024-12-05 115353

**Comment by JeremyParish69** [comment]
> So what is the schema to add an additional channel because there is a new client id, channel, and connection id for the transfer/transfer between migaloo<>terra

In line with my original think, previous cases where a wasm. IBC transfer port is registered as an additional channel, it shares the same clients and connection ids as the original transfer/transfer channel.

I'm not totally sure how to proceed, here. Either: 
- 1) this new channel (and the assets associated with) cannot be registered because it's not using the canonical IBC connection-ids and clients arleady registered for this chain pair--OR,
 - 2) if we can show that the transfer/transfer channel's connection and clients were created first, in use first, and have more transfer volume, then we can work to de-list the assets associated with the existing wasm. connection, and remove that wasm. channel from the registry, but that would be a big effort, since registering an IBC connection is generally considered permanent.

**Comment by calabiyau87** [comment]
So no way of registering a migaloo-terra_2.json with all the details mentioned above?

**Comment by JeremyParish69** [comment]
> So no way of registering a migaloo-terra_2.json with all the details mentioned above?

Alas, no, there's not

**Pull request merged**